### ISSUE TITLE
Fix hidden dependency on Magento_Multishipping

### ISSUE THREAD

**Issue by Stoux** [paynl/magento2-plugin#177]
There is currently a hidden dependency on `Magento_Multishipping` even though it's not defined in the `composer.json`. This causes build issues for projects that do not include Multishipping or have it replaced.

## Issue

When running the compile command (`bin/magento setup:di:compile`) build will fail on multiple files:
- [Model/PlaceMultiShippingOrder.php](https://github.com/paynl/magento2-plugin/blob/master/Model/PlaceMultiShippingOrder.php) references `Magento\Multishipping\Model\Checkout\Type\Multishipping\PlaceOrderInterface` which doesn't exist
- [Controller/Checkout/Finish.php](https://github.com/paynl/magento2-plugin/blob/master/Controller/Checkout/Finish.php) references `Magento\Multishipping\Model\Checkout\Type\Multishipping\State` which doesn't exist

You should be able to replicate it by adding a replace section to your project's root `composer.json`:

```
...
"replace": {
  "magento/module-multishipping": "*",
}
...
```

## Fix

We've "fixed" it by:
- Running `rm -f vendor/paynl/magento2-plugin/Model/PlaceMultiShippingOrder.php` in the post install script of composer
- Patching the line in `Finish.php`: `use Magento\Framework\DataObject as State; # PATCHED | Nasty... | Removed dependency on Magento\Multishipping\Model\Checkout\Type\Multishipping\State` 

Would love to see this handled as an actual soft dependency. Thanks in advance!

**Comment by escruickshank** [comment]
Hi Leon, thank you for reporting this! We will look into it.

**Comment by escruickshank** [comment]
Hi @Stoux, thank you for your comment. We have now fixed this in v3.5.1.
